sihl_contrib/session/lib/middleware.ml

```diff
@@ -1,11 +1,51 @@
-(* TODO
-https://docs.djangoproject.com/en/3.0/topics/http/sessions/
-*)
+let ( let* ) = Lwt.bind
 
-let m app =
+let cookie_key = "sessions_session_id"
+
+let session () app =
   let filter handler req =
-    Logs.warn (fun m -> m "session middleware is not implemented");
-    handler req
+    let (module Repository : Repo_sig.REPOSITORY) =
+      Sihl.Core.Registry.get Bind.Repository.key
+    in
+    match Opium.Std.Cookie.get req ~key:cookie_key with
+    | Some session_key -> (
+        let* session =
+          Repository.get ~key:session_key |> Sihl.Core.Db.query_db_exn req
+        in
+        match session with
+        | Some _ ->
+            (* TODO
+               1. fetch session data
+               2. decode to (string * string) list
+               3. put into env
+            *)
+            handler req
```

Comment:
Shouldn't the validity of the session be checked here? I know that the cookie expires when the session is expected to be expired, but maybe we should still double check. What you think? On the other hand, when session is refreshed, we should also remember to update the expiration time stored in the repo.

Reply:
It should be checked here, true. These are anonymous sessions and there is no concept of user, this comes when the user app is installed. I think anonymous sessions we should not refresh but just create a new one. The user app will base its sessions on top of these sessions, which will require a user to re-authenticate once the session expires.